velox/functions/sparksql/MightContain.cpp

```diff
@@ -64,8 +66,22 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> mightContainSignatures() {
               .build()};
 }
 
-std::unique_ptr<exec::VectorFunction> makeMightContain() {
-  return std::make_unique<BloomFilterMightContainFunction>();
+std::shared_ptr<exec::VectorFunction> makeMightContain(
+    const std::string& name,
+    const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  VELOX_CHECK_EQ(inputArgs.size(), 2);
+  BaseVector* serialized = inputArgs[0].constantValue.get();
+  if (!serialized) {
+    VELOX_USER_FAIL(
+        "{} requires first argument to be a constant of type VARBINARY", name);
+  }
+  BloomFilter bloomFilter;
+  if (!serialized->isNullAt(0)) {
```

Comment:
Can this be null? I assume no. Let's make this a VELOX_CHECK

Reply:
@jinchengchenghh Can you help to answer this question? If we add this check the test "nullBloomFilter" in MightContainTest will fail.

Reply:
Spark might_contain return null if any of input argument is null.
https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/BloomFilterAggregateQuerySuite.scala#L363

Previously, we use `isDefaultNullBehavior = true` to treat null inputs, but now, it cannot take effect.
So we need to check if it is null.

Reply:
Are you saying that the framework ends up creating function instance even when there is a constant input that is null? Looks like a bug to me. Would you file GitHub issue about it? CC: @laithsakka @Yuhta @bikramSingh91 

For now then, let's return an instance of AlwaysFailingVectorFunction when input is null.

Reply:
I am a bit lost here, why shall we create AlwaysFailingVectorFunction if expected behaviour is to return null?

Reply:
@mbasmanova  No, it's normal to check null when creating function instance because input may be null.

Reply:
@mbasmanova I agree with Chengcheng. https://github.com/apache/spark/blob/b496edb5d476b3ec13b5c69f3ab58b81e603e212/sql/core/src/test/scala/org/apache/spark/sql/BloomFilterAggregateQuerySuite.scala#L358-L372 In Spark's UT the bloom_filter_agg can produce null output with empty input, and in this case the input of "might_contain" is null. Here, the "might_contain" need to keep the same behavior as a sparksql function. I think this commit should be reverted https://github.com/facebookincubator/velox/pull/4728/commits/c12561d5294e21f810e977bc92cda768d8e25475

Reply:
@marin-ma Thank you for clarifying. I agree that we should make sure the semantics are identical to Spark's.